### PR TITLE
feat(android-signing-binskim): Relocate unified packing to signed build

### PR DESCRIPTION
**Pull request by jalkire**
#### Details
Currently, our unsigned release build builds the unified product, packs it into installers, and publishes those as artifacts. The signed build then picks up those installer artifacts and signs them. This PR relocates the packing step to the signed build. Now, the unsigned build builds the unified product and publishes the unpacked builds as artifacts. The signed build picks up those unpacked artifacts, packs them into installers, and signs them. There is no net effect on the unsigned + signed build combination.

##### Motivation
We currently only sign the packed installers. Now that we want to sign the contents of those installers, we will need to sign the unpacked builds, pack them, and then sign the packed installers. As such, packing needs to occur in the signed build.

##### Context
This PR follows up #4138. The next PR will actually add signing of the unpacked files, which will occur between the artifact download task and the packing task in the signed build.

There were a few alternate approaches considered. I could have combined our unsigned and signed builds, but that would have made it inconvenient to run debug builds without generating wasteful signed artifacts. I considered going in the other direction and adding extra builds, like unsigned build -> sign unpacked build -> packing build -> sign packed build, but that just seemed unnecessarily complicated. I also considered still packing and publishing installers in the unsigned build for debugging purposes, but that is easy to add later if desired so I left it off. 

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [n/a] Added/updated relevant unit test(s) (and ran `yarn test`)
- [n/a] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: pipeline/unified/publish-unpacked-build-output.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Licensed under the MIT License.
+parameters:
+    unpackedOutputPath: null
+    artifactName: null
+
+steps:
+    - task: CopyFiles@2
+      displayName: 'copy package files to upload'
+      inputs:
+          SourceFolder: ${{ parameters.packedOutputPath }}
```
Comment (dbjorge) [inline pipeline/unified/publish-unpacked-build-output.yaml:11]:
```suggestion
          SourceFolder: ${{ parameters.unpackedOutputPath }}
```

File: pipeline/unified/publish-unpacked-build-output.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+parameters:
+    unpackedOutputPath: null
+    artifactName: null
+
+steps:
+    - task: CopyFiles@2
+      displayName: 'copy package files to upload'
```
Comment (dbjorge) [inline pipeline/unified/publish-unpacked-build-output.yaml:9]:
```suggestion
      displayName: 'copy unpacked build output to artifact staging directory'
```

File: pipeline/unified/channel/sign-release-package-windows.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ jobs:
           unsignedPipelineResource: ${{ parameters.unsignedPipelineResource }}
           unsignedArtifactName: ${{ parameters.unsignedArtifactName }}
           signedArtifactName: ${{ parameters.signedArtifactName }}
+          channel: ${{ parameters.channel }}
```
Comment (dbjorge) [inline pipeline/unified/channel/sign-release-package-windows.yaml:14]: Should this be added as a parameter of this template?

**Review by dbjorge — CHANGES_REQUESTED** [review]
Thanks for breaking this out into multiple PRs! A few comments inline.

File: pipeline/unified/channel/sign-release-package-windows.yaml (hunk 3 above)
Comment (jalkire) [reply pipeline/unified/channel/sign-release-package-windows.yaml:15]: Good catch, yes!

**Comment by dbjorge** [comment]
LGTM now, but holding off approval while John works through an issue he hit in a test signing build

**Pull request closed without merging**